verifier/src/lib.rs

```diff
@@ -23,11 +23,67 @@ use mc_sgx_core_types::{
     Attributes, ConfigId, ConfigSvn, CpuSvn, ExtendedProductId, FamilyId, IsvProductId, IsvSvn,
     MiscellaneousSelect, MrEnclave, MrSigner, ReportData,
 };
-use subtle::{Choice, CtOption};
+use subtle::Choice;
 
 /// Number of spaces to indent nested error messages.
 const ERROR_INDENT: usize = 2;
 
+/// The output of a verification operation, [`Verifier::verify()`].
+#[derive(Debug, Clone)]
+pub struct VerificationOutput<T> {
+    // Whether or not this verification was successful.
+    succeeded: Choice,
+    // The value that was used in the verification.
+    value: T,
+}
+
+impl<T> VerificationOutput<T> {
+    /// Create a new instance.
+    ///
+    /// # Arguments
+    /// * `value` - The value that was used in the verification.
+    /// * `succeeded` - Whether or not the verification succeeded.
+    pub fn new(value: T, succeeded: Choice) -> VerificationOutput<T> {
+        Self { value, succeeded }
+    }
+
+    /// Was the verification successful?
+    pub fn is_success(&self) -> Choice {
+        self.succeeded
+    }
+
+    /// Was the verification a failure?
+    pub fn is_failure(&self) -> Choice {
+        !self.succeeded
+    }
+}
+
+impl<T> VerificationOutput<T>
+where
+    T: DisplayableError,
+{
+    /// Format the instance with preceding padding
+    ///
+    /// The `pad` is the number of spaces to precede each line of the displayed
+    /// representation with.
+    pub fn fmt_padded(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result {
+        match self.succeeded.unwrap_u8() {
+            1 => write!(f, "{:pad$}Passed", "")?,
+            _ => self.value.fmt_padded(f, pad)?,
+        }
+        Ok(())
+    }
+}
+
+impl<T> Display for VerificationOutput<T>
+where
+    T: DisplayableError,
+{
+    fn fmt(&self, f: &mut Formatter) -> core::fmt::Result {
+        self.fmt_padded(f, 0)
+    }
+}
+
```

Comment:
This and the fact that all of the verifier implementations return a `VerificationError` is a place holder for a follow on PR that will actually store just the value used during verification.